source/OrderedTypes/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,2209 @@
+
+\end{code}
+
+We now show that reindexing across an equivalence does not change the join.
+
+\begin{code}
+
+module equivalent-families-have-same-join {𝓤 𝓦 𝓥 𝓣 𝓣' : Universe}
+                                          (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                                          (T : 𝓣  ̇)
+                                          (T' : 𝓣'  ̇)
+                                          (e : T' ≃ T)
+                                          (m : T → ⟨ L ⟩)
+                                          where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+
+ reindexing-along-equiv-＝-sup : (s s' : ⟨ L ⟩)
+                               → (s is-lub-of (T , m)) holds
+                               → (s' is-lub-of (T' , m ∘ ⌜ e ⌝ )) holds
+                               → s ＝ s'
+ reindexing-along-equiv-＝-sup s s' (is-upbnd , is-least-upbnd)
+                        (is-upbnd' , is-least-upbnd') =
+   is-antisymmetric-for L s-≤-s' s'-≤-s
+  where
+   s-≤-s' : (s ≤ s') holds
+   s-≤-s' = is-least-upbnd (s' , λ t → transport (λ z → (z ≤ s') holds)
+                           (＝₁ t) (is-upbnd' (⌜ e ⌝⁻¹ t)))
+    where
+     ＝₁ : (t : T) → m (⌜ e ⌝ (⌜ e ⌝⁻¹ t)) ＝ m t
+     ＝₁ t = ap m (naive-inverses-are-sections ⌜ e ⌝ (⌜ e ⌝-is-equiv) t)
+   s'-≤-s : (s' ≤ s) holds
+   s'-≤-s = is-least-upbnd' (s , λ t' → is-upbnd (⌜ e ⌝ t'))
+
+\end{code}
+
+We can weaken the above result to a surjection.
+
+\begin{code}
+
+module surjection-implies-equal-joins {𝓤 𝓦 𝓥 𝓣 𝓣' : Universe}
+                                      (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                                      (T : 𝓣  ̇)
+                                      (T' : 𝓣'  ̇)
+                                      (e : T' ↠ T)
+                                      (m : T → ⟨ L ⟩)
+                                      where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+ open PropositionalTruncation pt
+
+ reindexing-along-surj-＝-sup : (s s' : ⟨ L ⟩)
+                              → (s is-lub-of (T , m)) holds
+                              → (s' is-lub-of (T' , m ∘ ⌞ e ⌟)) holds
+                              → s ＝ s'
+ reindexing-along-surj-＝-sup s s' (is-upbnd , is-least-upbnd)
+                        (is-upbnd' , is-least-upbnd') =
+   is-antisymmetric-for L s-≤-s' s'-≤-s
+  where
+   s-≤-s' : (s ≤ s') holds
+   s-≤-s' = is-least-upbnd (s' , λ t → map₂ t (⌞⌟-is-surjection e t))
+    where
+     map₁ : (t : T) → Σ t' ꞉ T' , ⌞ e ⌟ t' ＝ t → (m t ≤ s') holds
+     map₁ t (t' , path) = transport (λ z → (m z ≤ s') holds) path (is-upbnd' t')
+     map₂ : (t : T) → (Ǝ t' ꞉ T' , ⌞ e ⌟ t' ＝ t) holds → (m t ≤ s') holds
+     map₂ t = ∥∥-rec (holds-is-prop (m t ≤ s')) (map₁ t)
+   s'-≤-s : (s' ≤ s) holds
+   s'-≤-s = is-least-upbnd' (s , λ t' → is-upbnd (⌞ e ⌟ t'))
+
+\end{code}
+
+We now define a small basis for a sup-lattice as well as some boiler plate.
```

Comment:
The upcoming block should then go into a file called something like `SupLattice-SmallBasis`.